### PR TITLE
make the assembler more fast by avoiding useless empty sparse matrix creation

### PR DESCRIPTION
**Pull request by alessiofumagalli**
very easy branch, but I notices a speed up of 10 in the assembly procedure. Now most of the time is spent in the actual discretization and not in other stuff. 
Ciao ciao

### CODE REVIEW

File: src/porepy/numerics/mixed_dim/assembler.py [hunk 1]
```diff
@@ -120,12 +120,20 @@ def assemble_matrix_rhs(
             (g1, data_1, data_edge, local_matrix)
 
         """
+        # Define the matrix format, common for all the sub-matrices
+        if matrix_format == "csr":
+            sps_matrix = sps.csr_matrix
+        elif matrix_format == "csc":
+            sps_matrix = sps.csc_matrix
+        else: # coo
+            sps_matrix = sps.coo_matrix
```
Comment (keileg) [inline src/porepy/numerics/mixed_dim/assembler.py]: Are you sure coo is a good default value?

File: src/porepy/numerics/mixed_dim/assembler.py [hunk 2]
```diff
@@ -156,7 +164,13 @@ def assemble_matrix_rhs(
 
                             # Assign values in global matrix
                             var_key_name = self._variable_term_key(term, row, col)
-                            matrix[var_key_name][ri, ci] += loc_A
+                            # Check if the current block is None or not, it could
```
Comment (keileg) [inline src/porepy/numerics/mixed_dim/assembler.py]: Please explain why it can be none (initialization)

**Comment by alessiofumagalli** [comment]
note: the speed up was problem specific with 1 2d grid, 84 2d grid and 85 0d grid. but i guess it was mostly related to problems with many elements in the grid bucket

File: src/porepy/numerics/mixed_dim/assembler.py (hunk 1 above)
Comment (alessiofumagalli) [reply src/porepy/numerics/mixed_dim/assembler.py]: not really, should we put csr?

File: src/porepy/numerics/mixed_dim/assembler.py (hunk 2 above)
Comment (alessiofumagalli) [reply src/porepy/numerics/mixed_dim/assembler.py]: i just realized by running the tests

**Comment by alessiofumagalli** [comment]
python 3.6 works, 3.5 has the same problem as develop

**Pull request merged**